neo/rawio/intanrawio.py

```diff
@@ -496,39 +619,66 @@ def read_rhd(filename):
     ordered_channels = []
 
     if version >= V("1.2"):
-        data_dtype = [("timestamp", "int32", BLOCK_SIZE)]
+        if file_format == "header-attached":
+            data_dtype = [("timestamp", "int32", BLOCK_SIZE)]
+        else:
+            data_dtype[6] = [("timestamp", "int32",)]
+            channel_number_dict[6] = 1
     else:
-        data_dtype = [("timestamp", "uint32", BLOCK_SIZE)]
+        if file_format == "header-attached":
+            data_dtype = [("timestamp", "uint32", BLOCK_SIZE)]
+        else:
+            data_dtype[6] = [("timestamp", "uint32",)]
+            channel_number_dict[6] = 1
 
     # 0: RHD2000 amplifier channel
     for chan_info in channels_by_type[0]:
-        name = chan_info["native_channel_name"]
+        name = chan_info["custom_channel_name"]
         chan_info["sampling_rate"] = sr
         chan_info["units"] = "uV"
         chan_info["gain"] = 0.195
-        chan_info["offset"] = -32768 * 0.195
+        if file_format == "header-attached":
+            chan_info["offset"] = -32768 * 0.195
+        else:
+            chan_info["offset"] = 0.0
         ordered_channels.append(chan_info)
-        data_dtype += [(name, "uint16", BLOCK_SIZE)]
+        if file_format == "header-attached":
+            data_dtype += [(name, "uint16", BLOCK_SIZE)]
+        elif file_format == 'one-file-per-signal':
+            data_dtype[0] = "int16"
+        else:
+            data_dtype[0] += [(name, "int16")]
```

Comment:
I don't think you need the structured data type for the "one-file-per-channel" format either. 

Getting rid of this will allow you to ommit the overspecification of the channel in the `get_analog_signal_chunk` function:

https://github.com/zm711/python-neo/blob/839060e5b910d878021348de34f50e7bf1ce343e/neo/rawio/intanrawio.py#L290-L294

There, you first access the channel data with the index (that is, you know the channel already) but then, because is a structured datatype you need to index again with the channel name.

Reply:
True. You're right. I just really like naming memmaps. For no reason though. It's a strong point. I'll fix it.

Reply:
Made this change.